Service/ResponseTransformerStrategy/BadGatewayResponseStrategy.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Auto1\ServiceAPIClientBundle\Service\ResponseTransformerStrategy;
+
+use Auto1\ServiceAPIClientBundle\Service\ResponseTransformerStrategy;
+use Auto1\ServiceAPIComponentsBundle\Service\Endpoint\EndpointInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpFoundation\Response;
+use Auto1\ServiceAPIClientBundle\DTO\ResponseTransformer\ErrorResponse;
+use Auto1\ServiceAPIClientBundle\Exception\Response\BadGatewayResponseException;
+
+class BadGatewayResponseStrategy implements ResponseTransformerStrategy, LoggerAwareInterface
+{
+    use LoggerAwareTrait;
+
+    public function __construct()
+    {
+        $this->setLogger(new NullLogger());
+    }
+
+    public function supports(ResponseInterface $response): bool
+    {
+        return $response->getStatusCode() === Response::HTTP_BAD_GATEWAY;
+    }
+
+    public function handle(
```

Comment:
Lack of return type.

Reply:
no return type, as 'object|object[]|string' is an union type allowed from php8.0, so it will be not compatible with php7 when added to code, I added a @return doctype with it